shared/AppInsightsCommon/src/StorageHelperFuncs.ts

```diff
@@ -5,19 +5,24 @@ import {
     IDiagnosticLogger, _eInternalMessageId, _throwInternal, dumpObj, eLoggingSeverity, getExceptionName, getGlobal, getGlobalInst,
     isNullOrUndefined, objForEachKey
 } from "@microsoft/applicationinsights-core-js";
+import { ICachedValue, createCachedValue, objGetOwnPropertyDescriptor } from "@nevware21/ts-utils";
 import { StorageType } from "./Enums";
 
 let _canUseLocalStorage: boolean = undefined;
 let _canUseSessionStorage: boolean = undefined;
 let _storagePrefix: string = "";
 
+// Create cached values for verified storage objects to avoid repeated checks
+const _verifiedLocalStorage: ICachedValue<Storage> = createCachedValue<Storage>(null);
```

Comment:
Change this to be assigned to null by default, so that checking for a request is then a simple check to see if this has been evaluated yet